framework/console/controllers/BaseMigrateController.php

```diff
@@ -287,7 +288,7 @@ public function actionRedo($limit = 1)
         if (empty($migrations)) {
             $this->stdout("No migration has been done before.\n", Console::FG_YELLOW);
 
-            return self::EXIT_CODE_NORMAL;
+            return ExitCode::OK;
```

Comment:
asJson() is actually doing something, `exitOk()` would only return a fixed value. Also considering the amount of different values to choose from for an exit code I do not think a method makes sense. You can define your own exit code constants easily for returning them, defining a method for each of them does not make sense imo.